lib/pure/strmisc.nim

```diff
@@ -85,6 +86,139 @@ proc rpartition*(s: string, sep: string): (string, string, string)
+          bail("not 3 successive digits before decimal point, despite larger 1000.")
+        else:
+          when decimalDot != '.':
+            s[idx] = '.'  # Replace decimalDot to '.' so parseFloat can take it.
+          successive = 0
+          lastWasDot = true
+          afterDot = true
+          inc idx
+      of '-':  # Allow negative float
+        if unlikely(isNegative):  # Wont allow ---1.0
+          bail("string must not contain more than 1 '-' character.")
+        elif unlikely(idx != 0):
+          bail("the '-' character can only be at the start of the string.")
+        else:
+          isNegative = true
+          inc idx
+      else:
+        bail("invalid character in float: " & $s[idx])
+  parseFloat(s)
+
+
+func parseFloatThousandSep*(s: string; sep: static char = ','; decimalDot: static char = '.'): float {.since: (1, 3).} =
+  ## Convenience func for `parseFloat` which allows for thousand separators,
+  ## this is designed to parse floats as found in the wild formatted for humans.
+  ##
+  ## The following assumptions and requirements must be met:
+  ## - String must not be empty.
+  ## - String must be stripped of trailing and leading whitespaces.
+  ## - `sep` must not be `'-'` nor `' '`.
+  ## - `decimalDot` must not be `'-'` nor `' '`.
+  ## - `sep` and `decimalDot` must be different.
+  ## - No separator before a digit.
+  ## - First separator can be anywhere after first digit, but no more than 3 characters.
+  ## - There has to be 3 digits between successive separators.
+  ## - There has to be 3 digits between the last separator and the decimal dot.
+  ## - No separator after decimal dot.
+  ## - No duplicate separators.
+  ## - Floats without separator allowed.
+  ##
+  ## See also:
+  ## * `strutils <strutils.html>`_
+  runnableExamples:
+    doAssert parseFloatThousandSep("0") == 0.0
+    doAssert parseFloatThousandSep("-0") == -0.0
+    doAssert parseFloatThousandSep("0.0") == 0.0
+    doAssert parseFloatThousandSep("1.0") == 1.0
+    doAssert parseFloatThousandSep("-0.0") == -0.0
+    doAssert parseFloatThousandSep("-1.0") == -1.0
+    doAssert parseFloatThousandSep("1.000") == 1.0
+    doAssert parseFloatThousandSep("-1.000") == -1.0
+    doAssert parseFloatThousandSep("1,000") == 1000.0
+    doAssert parseFloatThousandSep("-1,000") == -1000.0
+    doAssert parseFloatThousandSep("10,000.000") == 10000.0
+    doAssert parseFloatThousandSep("1,000,000.000") == 1000000.0
+    doAssert parseFloatThousandSep("10,000,000.000") == 10000000.0
+    doAssert parseFloatThousandSep("10.000,0", '.', ',') == 10000.0
+    doAssert parseFloatThousandSep("1'000'000,000", '\'', ',') == 1000000.0
+  var copiedString = s
+  parseFloatThousandSepImpl(copiedString, sep, decimalDot)
+
+
+func parseFloatThousandSep*(s: var string; sep: static char = ','; decimalDot: static char = '.'): float {.since: (1, 3).} =
+  ## In-place version of `parseFloatThousandSep`, does not copy the string internally.
```

Comment:
Why does parsing require an internal copy? Remove this overload and fix your parsing code instead.